pkg/build/config.go

```diff
@@ -17,10 +17,16 @@ func GenerateConfigFromLabels(image string, config *api.Config) error {
 	}
 
 	var source *dockerclient.Image
-	if config.ForcePull {
+
+	switch config.BuilderPullPolicy {
+	case api.PullAlways:
 		source, err = d.PullImage(image)
-	} else {
+	case api.PullIfNotPresent:
 		source, err = d.CheckAndPullImage(image)
+	case api.PullNever:
+		source, err = d.CheckImage(image)
+	default:
```

Comment:
That default should not happen. IIRC, we have a validation method (see `pkg/api/validation`) which is being called before running a build. You should create a validation there.